File: 3634. Minimum Removals to Balance Array.cpp
class Solution {
public:
    int minRemoval(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int left = 0;
        int n = nums.size();
        int maxSize = 1;

        for (int right = 1; right < n; right++) {
            while (left < right && (long long)nums[left] * k < nums[right]) {
                left++;
            }
            maxSize = max(maxSize, right - left + 1);
        }

        return n - maxSize;
    }
};

File: 3634. Minimum Removals to Balance Array.java
class Solution {
    public int minRemoval(int[] nums, int k) {
        Arrays.sort(nums);

        int left = 0;
        int ans = 0;
        for(int right = 0; right < nums.length; right++){
            while(nums[right] > (long)nums[left] * k && left < right) left++;

            ans = Math.max(ans, right - left + 1);
        }
        // ans = Math.max(ans, nums.length - left - 1);

        return nums.length - ans;
    }
}